### DISCUSSION TITLE
is it possible to synch snapshots? or update children components after parents?

### DISCUSSION THREAD

**Discussion by KnightSlayer** [pmndrs/valtio#403] [Q&A]
Imagine that i have two nested components

``` js
const userService = proxy({...});

const ParentComponent = () => {
  const { user } = useSnapshot(userService);
  if (user) return <ChildComponent /> // render ChildComponent only if we have user
  return "Loading..."
}

const ChildComponent = () => {
  const { user } = useSnapshot(userService);
  return user.first_name; // in some circumstances i got error "Cannot read properties of null (reading 'first_name')"
}
```

Are there some neat solutions for this problem? (without props deilling)

**Comment by dai-shi** [comment]
If I understand it correctly, this is a batching issue. It will be resolved with React 18 automatic batching.
In React 17, we'd need to use `unstable_batchedUpdates` from `react-dom`, but it would be much easier to do `user?.first_name` in the mean time.
